File: src/dataset/python/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sieve_v2.py
"""Finds primes numbers"""

def sieve(lim):
    """Implimentation of sieve of Eratosthenes"""
    if lim < 2:
        return []
    if lim == 2:
        return [2]
    
    # Use boolean array instead of set for better memory efficiency
    is_prime = [True] * (lim + 1)
    is_prime[0] = is_prime[1] = False
    
    # Only check up to sqrt(lim)
    for num in range(2, int(lim**0.5) + 1):
        if is_prime[num]:
            # Mark multiples starting from num*num with step num
            for multiple in range(num * num, lim + 1, num):
                is_prime[multiple] = False
    
    # Collect primes in one pass
    return [num for num in range(2, lim + 1) if is_prime[num]]

File: src/dataset/javascript/sieve_exercism-ThomasZumsteg/claude/ClaudeSonnet4_sieve_v2.js
var Sieve = function(size) {
	/* Finds prime numbers */
	
	if (size < 2) {
		this.primes = [];
		return;
	}
	
	this.primes = [2];
	
	// Use typed array for better memory efficiency and performance
	var sieve = new Uint8Array(size + 1);
	sieve.fill(1);
	sieve[0] = sieve[1] = 0;
	
	// Only check odd numbers, skip even numbers after 2
	for(var n = 3; n * n <= size; n += 2) {
		if(sieve[n]) {
			// Start marking from n*n, skip even multiples
			for(var m = n * n; m <= size; m += 2 * n) {
				sieve[m] = 0;
			}
		}
	}
	
	// Collect remaining primes
	for(var i = 3; i <= size; i += 2) {
		if(sieve[i]) {
			this.primes.push(i);
		}
	}
}

module.exports = Sieve;